### ISSUE TITLE
Non-ASCII-8BIT special characters are being replaced with blank characters (ADDON-33494)

### ISSUE THREAD

**Issue by BartekZm** [splunk/splunk-connect-for-kubernetes#568]
**What happened**:
Add-on is replacing non-ASCII-8BIT special characters (like swedish, polish, russian, japaneese etc) with blank characters (or '?' in internal logs).

Example:
2021-02-26T08:18:29.227618927Z stdout F 2021-02-26 08:18:29 +0000 [info]: aaaąąąbbbźźźcccżżżddd

Is visible in Splunk in this way:
2021-02-26T08:18:29.227618927Z stdout F 2021-02-26 08:18:29 +0000 [info]: aaa   bbb   ccc   ddd
(GitHub is replacing 3 blank spaces with one, but I hope you feel the vibe)
Add-on container internal logs that are processing this event in the following way:
(...) 2021-02-26T08:18:29.227618927Z stdout F 2021-02-26 08:18:29 +0000 [info]: aaa???bbb???ccc???ddd

**What you expected to happen**:
Non-ASCII-8BIT characters should be visible and processed properly.

**How to reproduce it (as minimally and precisely as possible)**:
(in pod)
echo '2021-02-26T08:18:29.227618927Z stdout F 2021-02-26 08:18:29 +0000 [info]: aaaąąąbbbźźźcccżżżddd' >> name-splunk-kubernetes-logging-(POD).log

**Anything else we need to know?**:
Fix in pull request #567 - Change default encoding to UTF-8

**Workaround**:
1. Clone the main Splunk Connect for Kubernetes repo:
git clone https://github.com/splunk/splunk-connect-for-kubernetes.git
2. In 'configMap.yaml' that is located in the Helm chart folder, add 2 encoding statements - encoding + from_encoding - with proper libraries (written in capital letters), just after line 75 - <source> (https://github.com/splunk/splunk-connect-for-kubernetes/blob/79d17645af594762c9421cc184d59201a7ccb0a6/helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/configMap.yaml#L75):
vi splunk-connect-for-kubernetes/helm-chart/splunk-connect-for-kubernetes/charts/splunk-kubernetes-logging/templates/configMap.yaml

```
(...)
<source>
encoding UTF-8
from_encoding UTF-8
@id containers.log
(...)
```

3. Perform helm install, by using the local project content you modified:
helm install <NAME> -f <VALUES_FILE>.yaml splunk-connect-for-kubernetes/helm-chart/splunk-connect-for-kubernetes
4. All the non-native speakers' nightmare letters and sentences should be visualized properly in Splunk 
(like Grzegorz Brzęczyszczykiewicz. Chrząszczyrzewoszyce, powiat Łękołody
https://www.youtube.com/watch?v=AfKZclMWS1U).

**As per fluentd docs**:
https://docs.fluentd.org/input/tail#encoding-from_encoding
fluentd encoding can be set up by using the following parameters:
encoding
from_encoding
'- If encoding is specified, in_tail changes string to encoding.
This uses Ruby's String#force_encoding.
- If encoding and from_encoding both are specified, in_tail tries to
encode string from from_encoding to encoding. This uses Ruby's
- By default (it) emits string value as ASCII-8BIT encoding.'

**Environment**:
(environment is irrelevant for this issue TBH)
- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.2-34+350770ed07a558", GitCommit:"350770ed07a558ba1ce8202a6f9ea849f9a75f06", GitTreeState:"clean", BuildDate:"2021-02-11T20:09:31Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20+", GitVersion:"v1.20.2-34+350770ed07a558", GitCommit:"350770ed07a558ba1ce8202a6f9ea849f9a75f06", GitTreeState:"clean", BuildDate:"2021-02-11T20:10:29Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
- Ruby version (use `ruby --version`):
(in pod)
ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-linux]
- OS (e.g: `cat /etc/os-release`):
Ubuntu 20
- Splunk version:
8.1.2
- Splunk Connect for Kubernetes helm chart version:
1.4.4 - 1.4.6

**Comment by vinzent** [comment]
There is a similiar merged PR for the openshift logging component: https://github.com/openshift/origin-aggregated-logging/pull/622

They actually have a filter that runs just after reading any source and before more filters are run because not every source type supports setting encoding.

**Comment by rockb1017** [comment]
so if I add 
```
<filter **>
  @type record_modifier
  char_encoding utf-8
</filter>
```
right after the pipeline, would it be resolved? 
but i would have to make it optional with default to be off for performance issue, yea?

**Comment by vinzent** [comment]
my suggestion: https://github.com/splunk/splunk-connect-for-kubernetes/pull/570

**Comment by rockb1017** [comment]
@BartekZm Thank you for detailed instruction! I merged #570.
